From df0f64b0c9d35c8479a8b9258666872f2e2aca44 Mon Sep 17 00:00:00 2001
From: cym <cym@rock-chips.com>
Date: Sat, 2 Feb 2013 16:06:19 +0800
Subject: [PATCH] DDR:disable RK3168/RK3188 change DDR frequence when ddr_init,
 disable RK3168 change DDR frequence when suspend and video,because RK3168 DDR
 change frequence test fail.

---
 arch/arm/mach-rk30/board-rk3168-tb.c | 2 ++
 arch/arm/plat-rk/Kconfig             | 5 ++++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-rk30/board-rk3168-tb.c b/arch/arm/mach-rk30/board-rk3168-tb.c
index f709c23040d6..346a66723ae5 100755
--- a/arch/arm/mach-rk30/board-rk3168-tb.c
+++ b/arch/arm/mach-rk30/board-rk3168-tb.c
@@ -2033,8 +2033,10 @@ static struct cpufreq_frequency_table dvfs_gpu_table[] = {
 };
 
 static struct cpufreq_frequency_table dvfs_ddr_table[] = {
+#if defined(CONFIG_ARCH_RK3188)
 	{.frequency = 200 * 1000 + DDR_FREQ_SUSPEND,    .index = 950 * 1000},
 	{.frequency = 300 * 1000 + DDR_FREQ_VIDEO,      .index = 1000 * 1000},
+#endif
 	{.frequency = 396 * 1000 + DDR_FREQ_NORMAL,     .index = 1100 * 1000},
 	{.frequency = CPUFREQ_TABLE_END},
 };
diff --git a/arch/arm/plat-rk/Kconfig b/arch/arm/plat-rk/Kconfig
index 8c6ab2d1df30..725bfb6b8cb8 100755
--- a/arch/arm/plat-rk/Kconfig
+++ b/arch/arm/plat-rk/Kconfig
@@ -84,7 +84,9 @@ endchoice
 
 config DDR_INIT_CHANGE_FREQ
 	bool "Enable change DDR frequence when ddr_init"
-	default y
+	default y if ARCH_RK2928 || ARCH_RK30XX
+	default n if ARCH_RK3066B || ARCH_RK3188
+
 
 config DDR_SDRAM_FREQ
 	int "DDR SDRAM frequence (in MHz)"
@@ -93,6 +95,7 @@ config DDR_SDRAM_FREQ
 
 config DDR_FREQ
 	bool "Enable DDR frequency scaling"
+	default y if ARCH_RK3066B || ARCH_RK3188
 	select RK_SRAM_DMA if ARCH_RK30XX
 
 config DDR_TEST
-- 
2.35.3

